tracee/tracee.bpf.c

```diff
@@ -497,15 +498,26 @@ static __always_inline int get_config(u32 key)
     return *config;
 }
 
+// returns 1 if you should trace based on uid
 static __always_inline int uid_filter_matches()
 {
     uid_t uid = (u32)bpf_get_current_uid_gid();
-    u32* isMatch = bpf_map_lookup_elem(&uid_filter, &uid);
-    if (isMatch == NULL) {
-        return 0;
+    u32* equality = bpf_map_lookup_elem(&uid_equal_filter, &uid);
+    if (equality != NULL) {
+        return *equality;
```

Comment:
Why do you want to have that not trace anything? Would you expect `--filter uid=997,1000` to not produce any events? Would filters that just include '==' be OR'ed and otherwise AND'ed? 

That said I see the flaw in my current solution. Right now if someone were to specify `--filter 'uid>1' --filter 'uid<1000'` it wouldn't filter anything out. There are edge cases I appear to have overlooked.

For example, if a `!=` filter is specified every other UID should be included.

Reply:
> I think a user would expect that tracee --filter 'uid>5' --filter 'uid=1' would be logical ORed. I think a user would also expect tracee --filter uid>1 --filter uid<1000 to be logical ANDed. If you disagree, do we even want to trouble ourselves with supporting multiple filter specifications?

I agree. This is more intuitive. It just makes it a bit harder to implement - mixing AND and OR together.
Do you want to give it another try or maybe we should decide if we want to AND everything or OR everything? I think it will make the implementation easier